source/Octopus.Client/Repositories/BasicRepository.cs

```diff
@@ -156,6 +159,55 @@ public TResource Refresh(TResource resource)
             if (resource == null) throw new ArgumentNullException("resource");
             return Get(resource.Id);
         }
+
+        void EnrichSpaceIdIfRequire(TResource resource)
+        {
+            if (resource is IHaveSpaceResource spaceResource && TypeUtil.IsAssignableToGenericType(this.GetType(), typeof(ICanExtendSpaceContext<>)))
+            {
+                if (IsInSingleSpaceContext())
+                {
+                    spaceResource.SpaceId = Client.SpaceContext.SpaceIds.Single();
+                }
+            }
+        }
+
+        bool IsInSingleSpaceContext()
+        {
+            var spaceIds = GetSpaceIds();
+            var isASpecificSpaceId = spaceIds.Length == 1 && !spaceIds.Contains("all");
+            return isASpecificSpaceId && !IsIncludingSystem();
+        }
+
+        string[] GetSpaceIds()
+        {
+            var isMixedScope = TypeUtil.IsAssignableToGenericType(this.GetType(), typeof(ICanExtendSpaceContext<>));
+            return isMixedScope ? AdditionalQueryParameters["spaces"] as string[] : Client.SpaceContext.SpaceIds.ToArray();
+        }
+
+        bool IsIncludingSystem()
+        {
+            var isMixedScope = TypeUtil.IsAssignableToGenericType(this.GetType(), typeof(ICanExtendSpaceContext<>));
+            return isMixedScope
+                ? bool.Parse(AdditionalQueryParameters["includeSystem"].ToString())
+                : Client.SpaceContext.IncludeSystem;
+        }
+
+        void ValidateSpaceId(TResource resource)
+        {
+            if (resource is IHaveSpaceResource spaceResource)
+            {
+                var spaceIds = GetSpaceIds();
+                var isWildcard = spaceIds != null && spaceIds.Contains("all");
```

Comment:
another case of `"all"`

can you name this differently, we have a concept of wildcard in permissions, just so there's no way to confuse the concepts.